osbuild/pipeline.py

```diff
@@ -119,7 +119,7 @@ def __init__(self, runner=None, build=None):
 
     @property
     def id(self):
-        return self.output_id or self.tree_id
+        return self.output_id or self.tree_id or id(self)
```

Comment:
I think it would be helpful to have a comment on the `id` property. What is it meant to be used for / what properties is it required to have?

Reply:
This was a general musing, could be added (if you agree) in a follow-up.